src/main/scala/scalaz/stream/async/mutable/Queue.scala

```diff
@@ -45,10 +45,24 @@ trait Queue[A] {
    *
    * Please use `Topic` instead of `Queue` when all subscribers
    * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
    */
   def dequeue: Process[Task, A]
 
   /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
```

Comment:
Yeah, I would just bomb with an IllegalArgumentException in case of
negative values

On Mon, Mar 23, 2015 at 10:27 AM Alissa Pajer <notifications@github.com>
wrote:

> In src/main/scala/scalaz/stream/async/mutable/Queue.scala
> <https://github.com/scalaz/scalaz-stream/pull/342#discussion_r26940362>:
>
> >     */
> >    def dequeue: Process[Task, A]
> >
> >    /**
> > +   * Provides a process that dequeues in chunks.  Whenever *n* elements
> > +   * are available in the queue, `min(n, limit)` elements will be dequeud
> > +   * and produced as a single `Seq`.  Note that this naturally does not
> > +   * *guarantee* that `limit` items are returned in every chunk.  If only
> > +   * one element is available, that one element will be returned in its own
> > +   * sequence.  This method basically just allows a consumer to "catch up"
> > +   * to a rapidly filling queue in the case where some sort of batching logic
> > +   * is applicable.
> > +   */
> > +  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
>
> @pchiusano <https://github.com/pchiusano> yes you're right, I just found
> the spot. Down on line 185 we're checking (limit <= 0), and then taking
> all available elements if that's the case.
>
> I'd rather a dequeueAvailable function as well, at least in theory.
> Though the annoying thing with that solution, is that then in dequeueBatch,
> a limit < =0 case still must be handled. Though, I suppose in this case
> we could have the semantic that nothing is ever dequeued? Or we could just
> add an assertion that limit > 0 - not sure if assertions fit into this
> library?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scalaz/scalaz-stream/pull/342/files#r26940362>.
>